### PR TITLE
Solution transformStateWithClones

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -8,6 +8,42 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = { ...state };
+  const arrResult = [];
+
+  for (let i = 0; i < actions.length; i++) {
```
Comment (Anastasiia-Svintsova) [inline src/transformStateWithClones.js:14]: Let's use the for/of loop instead

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -8,6 +8,42 @@
...
+    if (actions[i]['type'] === 'addProperties') {
```
Comment (Anastasiia-Svintsova) [inline src/transformStateWithClones.js:15]: Use switch/case instead of `if`

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js (hunk 2 above)
Comment (Viktor-Kost) [reply src/transformStateWithClones.js]: is not fixed. Please apply recommendations

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Hi!
All nice, yet apply recommendation that previous mentor left (pay attention to unresolved conversation)

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -8,6 +8,43 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = { ...state };
+  const arrResult = [];
+
+  for (const action of actions) {
```
Comment (mykhalenych) [inline src/transformStateWithClones.js:14]: destruct your action

```suggestion
  for (const action of actions) {
  const { type, extraData, keysToRemove } = action;
```

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -8,6 +8,43 @@
...
+          };
+        };
+        break;
+
+      case 'clear':
+        for (const key in result) {
+          delete result[key];
+        };
+    };
+
```
Comment (mykhalenych) [inline src/transformStateWithClones.js:41]: add default case

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -8,6 +8,43 @@
...
+    const stage = { ...result };
+
+    arrResult.push(stage);
```
Comment (mykhalenych) [inline src/transformStateWithClones.js:44]:
```suggestion
    arrResult.push({ ...result });
```

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js [hunk 6]
```diff
@@ -8,6 +8,41 @@
...
+              delete result[key];
+            };
+          };
+        };
+        break;
+
+      default:
+        for (const key in result) {
+          delete result[key];
+        };
```
Comment (tetiana-ivanchenko) [inline src/transformStateWithClones.js:39]: it's not default case - it's delete case, default case shold handle errors

File: src/transformStateWithClones.js [hunk 7]
```diff
@@ -8,6 +8,41 @@
...
+        break;
+
+      case 'removeProperties':
+
+        for (const toRemove of keysToRemove) {
+          for (const key in result) {
+            if (toRemove === key) {
+              delete result[key];
+            };
+          };
```
Comment (tetiana-ivanchenko) [inline src/transformStateWithClones.js:32]: redundant nested loop, you shouldn't check keys, by the task - this keys 100% is keys of state

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**